### PR TITLE
Updated open mapping options. Closes #216

### LINKED ISSUE #216: Direct tile access has been discontinued when using Mapquest
From piyushsh18@gmail.com

"My xlsform collecting geopoint and it also use "appearance=maps" in input tag but when I try to collect geopoint from devices when mapping SDK selected as Openstreet Map SDK and basemap as Mapquest Streets/ Satellite the it want load the Mapquest tiles in basemap and it says "As of july 11, 2016 direct tile access has been discontinued".

And when I select mapping SDK as Google Maps SDK and basemap as Streets/ Satellite/ terrain/ Hybrid and then try to collect geopoint from the form window it automatically disappear or crash and navigate me to ODK main page.

So kindly guide me how to configure or what setting has to be performed for using Geo widgets."

### PR DESCRIPTION
**Pull request by max2me**
* Removed MapQuest tiles option
* Added OSM, USGS TOPO, USGS Satellite, Stamen Terrain tiles

### CODE REVIEW

File: collect_app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -342,11 +342,12 @@
 <string name="map_sdk_selector_title">Mapping SDK</string>
 <string name="map_sdk_behavior">Mapping SDK</string>
 <string name="google_maps_sdk">Google Maps SDK</string>
-<string name="openstreetmap_sdk">OpenStreet Map SDK</string>
+<string name="openstreetmap_sdk">Other Map SDKs</string>
```
Comment (lognaturel) [inline collect_app/src/main/res/values/strings.xml]: This should still be OpenStreetMap SDK. The basemap options have changed but not the SDK (there shouldn't be a space between Street and Map).

File: collect_app/src/main/java/org/odk/collect/android/spatial/TileSourceFactory.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import org.osmdroid.tileprovider.tilesource.OnlineTileSourceBase;
+import org.osmdroid.tileprovider.tilesource.XYTileSource;
+
+/**
+  * This factory augments Tile sources provides additional tile sources, some of which
+  * are available in new version of osmdroid but not in currently used 4.2
+ */
+public class TileSourceFactory {
+    public static final OnlineTileSourceBase USGS_TOPO = new OnlineTileSourceBase(
+            "USGS National Map Topo",
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/spatial/TileSourceFactory.java]: I don't think these names are currently used in the interface but for consistency's sake, we should probably use the String resources.

**Review by lognaturel — CHANGES_REQUESTED** [review]
Sweet! Just a couple small things I marked inline.

Did you see a reason not to include the CartoDB basemaps with free licenses (Positron and Dark Matter)?

File: collect_app/src/main/res/values-de/strings.xml [hunk 3]
```diff
@@ -338,9 +338,6 @@
   <string name="google_maps_sdk">Google Maps SDK</string>
   <string name="openstreetmap_sdk">OpenStreetMap SDK</string>
   <string name="show_map_sdk">Map-SDK-Optionen anzeigen</string>
-  <string name="openstreetmap">OpenStreetMap</string>
```
Comment (lognaturel) [inline collect_app/src/main/res/values-de/strings.xml:341]: Don't worry about translations, only the strings.xml file matters. The others will get pulled from Transifex (https://www.transifex.com/opendatakit/odk-collect/) and overwritten.

**Comment by jnordling** [comment]
@max2me the pull request looks good, @lognaturel made some comments. I think if you make those small modification we can pull it in. 

It would be great to add this to master. Every little bit helps, Thanks!!

**Comment by lognaturel** [comment]
@jnordling Do you have any thoughts on the basemaps? Is it worth also including the free ones from CartoDB or are they mostly just redundant?

**Comment by jnordling** [comment]
Yeah I think as long as some the core tiles work. In general they are all the same except for some cartographic differences. However it would be sweet to have CartoDB and any others like MapBox free maps in there... The idea solution is to have a configurable setting in either admin or general settings to specify a custom URL to a tile server... Just an idea... Keeping basemap URL's up to-date is always challenging...

**Comment by lognaturel** [comment]
Ok! @max2me Can you please look into both CartoDB and MapBox free maps for now? And @jnordling, I like this configurable idea. How about you file it as an issue for now and we can discuss there what that could look like.

File: collect_app/src/main/res/values/strings.xml (hunk 1 above)
Comment (max2me) [reply collect_app/src/main/res/values/strings.xml]: @lognaturel I've changed it back to OpenStreetMap SDK

File: collect_app/src/main/java/org/odk/collect/android/spatial/TileSourceFactory.java (hunk 2 above)
Comment (max2me) [reply collect_app/src/main/java/org/odk/collect/android/spatial/TileSourceFactory.java]: I've refactored TileSourceFactory to now pull up those strings from resources (couldnt find how they are used either).

**Comment by max2me** [comment]
@lognaturel @jnordling In general my hesitation with both CartoDB and MapBox is that they impose limits on tile views per month. 

* CartoDB offers 75,000 (or 100,000 - what they have published on the site is different from what they have in [Terms of Service](https://drive.google.com/file/d/0B3OBExqwT6KJaHdTcEVnYU8xYTg/view)) for free, 
* MapBox's free plan includes 50,000 map tiles per month and up to 50,000 mobile users.

Also free tiles from CartoDb seem to be more suited for visualization and rendering data on top of them.

I'm not sure how CartoDB enforces that limit since they do not require API key in tile URL so I'm going to add those. Let me know if I should sign up for MapBox and add those layers as well. 

Another option would be to let users enter their own MapBox account key in settings, that way they will be less likely to be affected by limits imposed.

**Comment by max2me** [comment]
@lognaturel @jnordling Done. Please take a look.

**Comment by lognaturel** [comment]
Cool. One thing I hadn't noticed before is that the USGS maps seem to be of pretty limited use outside the US. Maybe we should have a more considered conversation about which basemaps should be included.

It looks like the CartoDB tiles share a cache or something. If I set the basemap as Positron, load a few tiles and then change it to Dark Matter, the tiles that were loaded as Positron don't get loaded again and so I have tiles of both kinds showing up.

@yanokwa I think we can merge this now even with those small issues so there are osm sdk basemaps available. Then we can continue with a broader conversation about how to improve mapping.

File: collect_app/src/main/java/org/odk/collect/android/spatial/TileSourceFactory.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+        mStamenTerrain = new XYTileSource(context.getString(R.string.openmap_stamen_terrain),
+                ResourceProxy.string.unknown, 0, 18, 256, ".jpg", new String[] {
+                "http://tile.stamen.com/terrain/" });
+
+        mCartoDbPositron = new XYTileSource(context.getString(R.string.openmap_cartodb_positron),
+                ResourceProxy.string.unknown, 0, 18, 256, ".png", new String[] {
+                "http://1.basemaps.cartocdn.com/light_all/" });
+
+        mCartoDbDarkMatter = new XYTileSource(context.getString(R.string.openmap_cartodb_positron),
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/spatial/TileSourceFactory.java]: @max2me This is what's causing the shared cache between the two CartoDB basemaps. Please fix!

Comment (lognaturel) [reply collect_app/src/main/java/org/odk/collect/android/spatial/TileSourceFactory.java]: To provide a bit of context, osmdroid saves tiles to the SDCard in the osmdroid/tiles folder. It makes a folder for each basemap using that string.

**Pull request merged**